index.js

```diff
@@ -18,9 +18,9 @@ module.exports = function(options) {
 function toPathMatch(pattern) {
   if (typeof pattern === 'string') {
     const reg = pathToRegexp(pattern, [], { end: false });
-    return ctx => reg.test(ctx.path);
```

Comment:
keep using `test()` match will be better, it won't store the match items.

```js
reg.lastIndex = 0;
return reg.test(ctx.path);
```

Just reset the `lastIndex` before runing test method.

Reply:
change 'match()' to 'test()', because it has better performance. [https://jsperf.com/match-vs-test/9](https://jsperf.com/match-vs-test/9)
![image](https://user-images.githubusercontent.com/5102113/32817620-6efdb330-c984-11e7-943d-930497e1aebc.png)
![image](https://user-images.githubusercontent.com/5102113/32817627-7b0b5c54-c984-11e7-8cef-ac79ede913cb.png)